aea/cli/utils/package_utils.py

```diff
@@ -70,11 +76,19 @@ def verify_or_create_private_keys(ctx: Context) -> None:
     for identifier, private_key_path in IDENTIFIER_TO_KEY_FILES.items():
         config_private_key_path = aea_conf.private_key_paths.read(identifier)
         if config_private_key_path is None:
-            create_private_key(identifier)
-            aea_conf.private_key_paths.update(identifier, private_key_path)
+            if identifier == aea_conf.default_ledger:
+                create_private_key(
+                    identifier,
+                    private_key_file=str(aea_project_path / private_key_path),
+                )
+                aea_conf.private_key_paths.update(identifier, private_key_path)
```

Comment:
Only the default ledger should get a key pair created. Otherwise there are dangerous side effects which are hard to debug.